Applications/Spire/Source/KeyBindings/CancelKeyBindingsTableModel.cpp

```diff
@@ -0,0 +1,171 @@
+#include "Spire/KeyBindings/CancelKeyBindingsTableModel.hpp"
+
+using namespace Spire;
+
+namespace {
+  const auto ROW_COUNT = 13;
+  const auto COLUMN_COUNT = 2;
+
+  auto get_action(int index) {
+    static auto actions = std::vector<KeyBindings::CancelAction>({
+      KeyBindings::CancelAction::ALL,
+      KeyBindings::CancelAction::ALL_ASKS,
+      KeyBindings::CancelAction::ALL_BIDS,
+      KeyBindings::CancelAction::CLOSEST_ASK,
+      KeyBindings::CancelAction::CLOSEST_BID,
+      KeyBindings::CancelAction::FURTHEST_BID,
+      KeyBindings::CancelAction::FURTHEST_ASK,
+      KeyBindings::CancelAction::MOST_RECENT,
+      KeyBindings::CancelAction::MOST_RECENT_ASK,
+      KeyBindings::CancelAction::MOST_RECENT_BID,
+      KeyBindings::CancelAction::OLDEST,
+      KeyBindings::CancelAction::OLDEST_ASK,
+      KeyBindings::CancelAction::OLDEST_BID
+    });
+    return actions[index];
+  }
+
+  auto get_index(KeyBindings::CancelAction action) {
+    static auto indexes = std::unordered_map<KeyBindings::CancelAction, int>({
+      {KeyBindings::CancelAction::ALL, 0},
+      {KeyBindings::CancelAction::ALL_ASKS, 1},
+      {KeyBindings::CancelAction::ALL_BIDS, 2},
+      {KeyBindings::CancelAction::CLOSEST_ASK, 3},
+      {KeyBindings::CancelAction::CLOSEST_BID, 4},
+      {KeyBindings::CancelAction::FURTHEST_BID, 5},
+      {KeyBindings::CancelAction::FURTHEST_ASK, 6},
+      {KeyBindings::CancelAction::MOST_RECENT, 7},
+      {KeyBindings::CancelAction::MOST_RECENT_ASK, 8},
+      {KeyBindings::CancelAction::MOST_RECENT_BID, 9},
+      {KeyBindings::CancelAction::OLDEST, 10},
+      {KeyBindings::CancelAction::OLDEST_ASK, 11},
+      {KeyBindings::CancelAction::OLDEST_BID, 12}});
+    return indexes[action];
+  }
+}
+
+CancelKeyBindingsTableModel::CancelKeyBindingsTableModel(
+    const std::vector<KeyBindings::CancelActionBinding>& bindings,
+    QObject* parent)
+    : QAbstractTableModel(parent) {
```

Comment:
In general, when you are about to fill a vector up to a pre-set size, construct the vector with that size right away. It is faster (no re-allocations occurred throughout the process) and can be less verbose too. When impossible (i.e. there are no zero-cost constructors for the element type), use the default constructor followed with a call to `std::vector::reserve`.
In this case, you can construct the vector with the size needed directly in the initializer list.